pkg/main.go

```diff
@@ -0,0 +1,94 @@
+package pkg
+
+import (
+	"time"
+
+	"github.com/v3io/registry-creds-handler/pkg/registry"
+	"github.com/v3io/registry-creds-handler/pkg/util"
+
+	"github.com/nuclio/errors"
+	"github.com/nuclio/logger"
+	"k8s.io/client-go/kubernetes"
+)
+
+type Handler struct {
+	logger        logger.Logger
+	kubeClientSet *kubernetes.Clientset
+	ecr           *registry.ECR
+	refreshRate   time.Duration
+	registryKind  string
+}
+
+func NewHandler(logger logger.Logger,
+	kubeClientSet *kubernetes.Clientset,
+	ecr *registry.ECR,
+	refreshRate int64,
+	registryKind string) (*Handler, error) {
+
+	return &Handler{
+		logger:        logger.GetChild("handler"),
+		kubeClientSet: kubeClientSet,
+		ecr:           ecr,
+		refreshRate:   time.Duration(refreshRate) * time.Minute,
+		registryKind:  registryKind,
+	}, nil
+}
+
+func (h *Handler) Run() error {
+	h.logger.Info("Handler starting...")
+	switch h.registryKind {
+	case registry.ECRRegistryKind:
+		err := h.createOrUpdateECRSecret()
+		if err != nil {
+			return errors.Wrap(err, "Failed to create or update ECR token secret")
+		}
+
+		// should never return
+		err = util.SyncSecret(h.kubeClientSet, h.refreshRate, h.ecr.Namespace, h.ecr.SecretName, h.createOrUpdateECRSecret)
+		if err != nil {
+			return errors.Wrap(err, "Failed to sync secret")
+		}
```

Comment:
use a for loop with a time ticker, run in a go routine.
it is a common practice to handle forever running go routines with a way to stop it. it will help you when you are testing your code (you can start/stop anytime).
also, make sure the caller provide a context and use that on your routine. once the context is cancelled, your ongoing process on routine would be stopped immediately and return with an error. use the `ctx` within your select to make sure it was not ended.

remember, errors occurs within the `for` should be logged and not return